qiita_db/util.py

```diff
@@ -2311,3 +2317,337 @@ def send_email(to, subject, body):
         raise RuntimeError("Can't send email!")
     finally:
         smtp.close()
+
+
+def resource_allocation_plot(file, cname, sname='all'):
+    """Builds resource allocation plot for given filename and jobs
+
+    Parameters
+    ----------
+    file : str, required
+        Builds plot for the specified file name. Usually provided as tsv.gz
+    cname: str, required
+        Specified job type
+    sname: str, optional
+        Specified job sub type.
+
+    Returns
+    ----------
+    matplotlib.pyplot object
+        Returns a matplotlib object with a plot
+    """
+    # Constants
+    COL_NAME = 'samples * columns'
+
+    df = pd.read_csv(file, sep='\t', dtype={'extra_info': str})
+    df['ElapsedRawTime'] = pd.to_timedelta(df.ElapsedRawTime)
+    if sname == "all":
+        _df = df[(df.cName == cname)].copy()
+    else:
+        _df = df[(df.cName == cname) & (df.sName == sname)].copy()
+
+    fig, axs = plt.subplots(ncols=2, figsize=(10, 4), sharey=False)
+    _df.dropna(subset=['samples', 'columns'], inplace=True)
+    _df[COL_NAME] = _df.samples * _df['columns']
+    ax = axs[0]
+    # k a b
+    # bmin = _df['MaxRSSRaw'].min()
+    models = [mem_model1, mem_model2, mem_model3, mem_model4]
+    _resource_allocation_plot_helper(_df, ax, cname, sname, "MaxRSSRaw",
+                                     models)
+
+    # bmin = _df['ElapsedRaw'].min()
+    ax = axs[1]
+    models = [time_model1, time_model2, time_model3, time_model4]
+    _resource_allocation_plot_helper(_df, ax, cname, sname, "ElapsedRaw",
+                                     models)
+    return fig, axs
+
+
+def _resource_allocation_plot_helper(_df, ax, cname, sname, curr, models):
+    COL_NAME = 'samples * columns'
+    M1G = 2**30
+    """Helper function for resource allocation plot. Builds plot for MaxRSSRaw
+    and ElapsedRaw
+
+    Parameters
+    ----------
+    _df: pandas dataframe, required
+        Filtered dataframe for the plot
+    ax : matplotlib axes, required
+        Axes for current subplot
+    cname: str, required
+        Specified job type
+    sname: str, optional
+        Specified job sub type.
+    curr: str, required
+        Either MaxRSSRaw or ElapsedRaw
+    models: list, required
+        List of functions that will be used for visualization
+
+    """
+
+    x_data, y_data = _df[COL_NAME], _df[curr]
+    ax.scatter(x_data, y_data, s=2, label="data")
+    d = dict()
+    for index, row in _df.iterrows():
+        x_value = int(row[COL_NAME])
```

Comment:
why not cast via pandas the full column vs. doing it on each iteration?

Reply:
I'm not sure how to create a dict of lists using pandas